### ISSUE TITLE
Edit segment screen fails to load for user without Manage segments permission

### ISSUE THREAD

**Issue by luk355** [SolidStateGroup/bullet-train-api#2073]
**Describe the bug**

User without _Manage segments_ permissions gets 500 API error when navigating to _Edit Segment_ screen.

**To Reproduce**

1. Create user without _Manage segment_ permission
2. Go to Segments -> Click on specific segment
3. Edit Segment page fails to load - 500 API error code can be seen in the network tab

**Expected behavior**

User can navigate to specific segment, the data are shown as read-only and cannot be modified.

**Screenshots**

![image](https://user-images.githubusercontent.com/9592890/227927957-3e78dcb4-e057-4e89-9c34-900e36aa7cc6.png)

**How are you running Flagsmith?**

- [X] Self Hosted with Kubernetes

**Additional context**

Image versions: flagsmith-api-ee:2.40.1, flagsmith-frontend:2.41.0
Stack trace:

```
AttributeError: 'Request' object has no attribute 'master_api_key'
return self.__getattribute__(attr)
File "/opt/venv/lib/python3.10/site-packages/rest_framework/request.py", line 418, in __getattr__
master_api_key = request.master_api_key
File "/app/src/segments/permissions.py", line 60, in has_object_permission
self.op2.has_object_permission(request, view, obj)
File "/opt/venv/lib/python3.10/site-packages/rest_framework/permissions.py", line 82, in has_object_permission
if not permission.has_object_permission(request, self, obj):
File "/opt/venv/lib/python3.10/site-packages/rest_framework/views.py", line 345, in check_object_permissions
self.check_object_permissions(self.request, obj)
File "/opt/venv/lib/python3.10/site-packages/rest_framework/generics.py", line 99, in get_object
instance = self.get_object()
File "/opt/venv/lib/python3.10/site-packages/rest_framework/mixins.py", line 54, in retrieve
response = handler(request, *args, **kwargs)
File "/opt/venv/lib/python3.10/site-packages/rest_framework/views.py", line 506, in dispatch
raise exc
File "/opt/venv/lib/python3.10/site-packages/rest_framework/views.py", line 480, in raise_uncaught_exception
self.raise_uncaught_exception(exc)
File "/opt/venv/lib/python3.10/site-packages/rest_framework/views.py", line 469, in handle_exception
response = self.handle_exception(exc)
File "/opt/venv/lib/python3.10/site-packages/rest_framework/views.py", line 509, in dispatch
return self.dispatch(request, *args, **kwargs)
File "/opt/venv/lib/python3.10/site-packages/rest_framework/viewsets.py", line 125, in view
return view_func(*args, **kwargs)
File "/opt/venv/lib/python3.10/site-packages/django/views/decorators/csrf.py", line 54, in wrapped_view
response = wrapped_callback(request, *callback_args, **callback_kwargs)
File "/opt/venv/lib/python3.10/site-packages/django/core/handlers/base.py", line 181, in _get_response
response = get_response(request)
File "/opt/venv/lib/python3.10/site-packages/django/core/handlers/exception.py", line 47, in inner
Traceback (most recent call last):
 
During handling of the above exception, another exception occurred:
 
AttributeError: 'WSGIRequest' object has no attribute 'master_api_key'
return getattr(self._request, attr)
File "/opt/venv/lib/python3.10/site-packages/rest_framework/request.py", line 416, in __getattr__
Traceback (most recent call last):
django.request ERROR    Internal Server Error: /api/v1/projects/1/segments/7/
```

**Comment by matthewelwell** [comment]
Thanks @luk355, I'll look into this for you and come back to you as soon as I can.

**Comment by matthewelwell** [comment]
Hey @luk355, thanks again for raising this - I have created a PR to resolve it here: #2074 . I will keep you posted on when it is added to the enterprise image as well.

**Comment by luk355** [comment]
Hey @matthewelwell, was this fix released by part of the enterprise image, yet? Thanks!

**Comment by matthewelwell** [comment]
Hey @luk355, This will be included in the Enterprise release at the end of this month.

**Comment by matthewelwell** [comment]
Hey @luk355, this was released in v2.51.1  of the flagsmith-api-ee image last week.
